components/kyma-environment-broker/internal/servicemanager/client.go

```diff
@@ -0,0 +1,243 @@
+package servicemanager
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/Peripli/service-manager-cli/pkg/query"
+	"github.com/Peripli/service-manager-cli/pkg/types"
+	"github.com/Peripli/service-manager/pkg/web"
+	kebError "github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/error"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/iosafety"
+	"github.com/pkg/errors"
+)
+
+type (
+	client struct {
+		creds      Credentials
+		httpClient *http.Client
+	}
+)
+
+func New(credentials Credentials) Client {
+	return &client{
+		creds:      credentials.WithNormalizedURL(),
+		httpClient: http.DefaultClient,
```

Comment:
I think it is worth to configure at least some timeouts on the HTTP client.

Reply:
good point, let's do a new constructor with http.Client as an argument and the factory, which creates sm client has one shared and configured http.Client